core-bundle/src/Controller/AbstractFragmentController.php

```diff
@@ -27,24 +28,48 @@ abstract class AbstractFragmentController extends AbstractController implements
      */
     protected $options = [];
 
-    /**
-     * @var PageModel|null
-     */
-    private $pageModel;
-
     public function setFragmentOptions(array $options): void
     {
         $this->options = $options;
     }
 
-    protected function setPageModel(?PageModel $pageModel): void
+    public static function getSubscribedServices(): array
     {
-        $this->pageModel = $pageModel;
+        $services = parent::getSubscribedServices();
+
+        $services['request_stack'] = RequestStack::class;
+
+        return $services;
     }
 
     protected function getPageModel(): ?PageModel
     {
-        return $this->pageModel;
+        $request = $this->get('request_stack')->getCurrentRequest();
+
+        if (null === $request || !$request->attributes->has('pageModel')) {
+            return null;
+        }
```

Comment:
The equivalent in the `ContaoContext` class looks like this:

```php
        if (null === $request || !$request->attributes->has('pageModel')) {
            if (isset($GLOBALS['objPage']) && $GLOBALS['objPage'] instanceof PageModel) {
                return $GLOBALS['objPage'];
            }

            return null;
        }
```

Do we not need the `$GLOBALS['objPage']` check here?

Reply:
No I don't think we want that in a controller. In a controller, the page model should always relate to the current request. If I call `$this->getPageModel()` in my controller method, but my request/route does not contain that information, I would expect to get a `null` value returned.